### ISSUE TITLE
Invalid types on HoneybadgerErrorBoundary and ErrorCompon

### ISSUE THREAD

**Issue by dvictory** [honeybadger-io/honeybadger-js#1075]
## What are the steps to reproduce this issue?
1. use Typescript
2. Create a React Component(DefaultErrorComponent), copy contents of DefaultErrorComponent (Honeybadger version) to component, basically make a copy
3. Pass the Component as the property ErrorComponent to HoneybadgerErrorBoundary

## What happens?
Type error CustomComponent is not of type ReactNode

## What were you expecting to happen?
No Type errors

## Any logs, error output, etc?
Does not compile.

## Any other comments?
I think the type defs for the HoneybadgerErrorBoundary are incorrect.  

https://github.com/honeybadger-io/honeybadger-js/blob/master/packages/react/src/HoneybadgerErrorBoundary.tsx#L22 conflicts with https://github.com/honeybadger-io/honeybadger-js/blob/master/packages/react/src/HoneybadgerErrorBoundary.tsx#L9

ReactNode is the first, and the 2nd is PropTypes.oneOfType([PropTypes.element, PropTypes.func]).

I am no React expert, but seems like something is off. I would think you could pass a copy of the default as an override and not have issues.

I solved with:
```jsx
HoneybadgerErrorBoundary
          honeybadger={Honeybadger}
          ErrorComponent={DefaultErrorComponent as unknown as ReactNode}
        >
```
ew, this seems wrong.

Thanks

**Comment by subzero10** [comment]
Hey @dvictory, thank you for submitting an issue!

Indeed, this `ErrorComponent={DefaultErrorComponent as unknown as ReactNode}` does not look good!
I tried reproducing your issue in the [example project](https://github.com/honeybadger-io/honeybadger-js/tree/master/packages/react/example), but it seems that a custom error component works for me.
Here's what I did:
- I created a `CustomErrorComponent.tsx`, which is a copy of the `DefaultErrorComponent.tsx`, with the class and file names changed.
- I pass the CustomErrorComponent to the `HoneybadgerErrorBoundary`:
```
  import CustomErrorComponent from './CustomErrorComponent';
  ...
  ReactDOM.render(
    <HoneybadgerErrorBoundary honeybadger={Honeybadger} ErrorComponent={CustomErrorComponent}>
      <App />
    </HoneybadgerErrorBoundary>,
    document.getElementById('root')
  )
```
- I start the project with `npm run start` and all works as expected.

---
I took a look at the type definitions:
`ReactNode` is a `ReactElement`
```
// @types/react/index.d.ts
type ReactChild = ReactElement | ReactText;
type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
```
and `PropTypes.element` takes an interface `ReactElementLike` which makes me think that it shouldn't error.

Can you please tell me the exact versions of:
- `@types/react` (mine is at v17.0.50)
- `@types/prop-types` (mine is at v15.7.5)

I suspect that the issue is that the type definitions of your installed packages are incompatible. If that's the case, I will make sure that the versions installed are compatible.

**Comment by dvictory** [comment]
Thanks, this is what I have;


@types/react = 18.0.25
@types/prop-types = 15.8.1

**Comment by subzero10** [comment]
@dvictory This should be fixed with v6.0.3!
